### PR TITLE
Add Default Error Message in Cases Where there are No Itins

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**

Adds a default OTP-style error message encouraging a user to change their search if we are returning a blank itinerary results pane. 

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I don't like the approach of putting this data decision in the presentation part of the error renderer. Could we instead dynamically add the error to the error set the way we do for `OTP_RR_A11Y_ROUTING_ENABLED`?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Also, could we change this from `DEFAULT_ERROR_MESSAGE` to something like `NO_ITINS_RETURNED`?

**Comment by amy-corson-ibigroup** [comment]
@miles-grant-ibigroup We can add this to the error set in cases where the we actually return a response from the server! The issue is when the query is invalid (such as when a user only selects "rental" in the mode selector), we don't actually dispatch the routing request, and so there's no error set to add to.

**Comment by miles-grant-ibigroup** [comment]
> @miles-grant-ibigroup We can add this to the error set in cases where the we actually return a response from the server! The issue is when the query is invalid (such as when a user only selects "rental" in the mode selector), we don't actually dispatch the routing request, and so there's no error set to add to.

That makes sense. But it definitely has to be added somewhere in the routing request process rather than at render time. What we have now is not ok

**Review by miles-grant-ibigroup — APPROVED** [review]
Beautiful work!  Excellently architected. This should be in a textbook

File: i18n/en-US.yml [hunk 1]
```diff
@@ -384,7 +384,7 @@ components:
       body: >-
         No transit connection was found between your origin and destination on
         the selected day of service, using the vehicle types you selected.
-      header: No transit connections
+      header: No itineraries found
```
Comment (binh-dam-ibigroup) [inline i18n/en-US.yml:387]: Do the other languages need to be updated too?

**Review by binh-dam-ibigroup — APPROVED** [review]

File: i18n/en-US.yml (hunk 1 above)
Comment (amy-corson-ibigroup) [reply i18n/en-US.yml:387]: I think so!

**Pull request merged**